### PR TITLE
Cookie compress using brotli

### PR DESCRIPTION
**Pull request by andrewshkovskii**

### CODE REVIEW

File: werkzeug_encryptedcookie/__init__.py [hunk 1]
```diff
@@ -66,6 +80,12 @@ def unserialize(cls, string, secret_key):
             except Exception:
                 pass
 
+        if cls.compress_cookie:
+            try:
+                string = cls.decompress(string)
+            except brotli.error:
```
Comment (homm) [inline werkzeug_encryptedcookie/__init__.py]: Brotli can successfully decompress random strings and return empty strings. Proof:

```python
import os, brotli
def test_b(): 
    s = os.urandom(89) 
    try: 
        brotli.decompress(s) 
    except brotli.error: 
        return 
    except: 
        pass 
    print('>>>', repr(s), repr(brotli.decompress(s))) 

%timeit test_b()
```

We need some kind of marker.

Comment (andrewshkovskii) [reply werkzeug_encryptedcookie/__init__.py]: Not sure if we should check for header or just check for `startswith` and then cur the `data`.
Maybe two checks for `separator` after `partition()` and check of result of decompression is enough. What do you think? @homm

File: test.py [hunk 2]
```diff
@@ -79,12 +82,23 @@ def test_fail_when_corrupted(self):
...
+
 
 class SecureEncryptedCookieTest(EncryptedCookieTest):
     Cookie = SecureEncryptedCookie
     class RawCookie(Cookie):
         quote_base64 = False
 
+    class RawCookie(Cookie):
+        compress_cookie = False
+
```
Comment (homm) [inline test.py]: ???

File: werkzeug_encryptedcookie/__init__.py [hunk 3]
```diff
@@ -73,6 +96,12 @@ def unserialize(cls, string, secret_key):
 
         payload = cls.decrypt(string, secret_key)
 
+        if cls.compress_cookie:
+            try:
+                payload = cls.decompress(payload)
+            except brotli.error:
```
Comment (homm) [inline werkzeug_encryptedcookie/__init__.py]: Exception logic looks better in the method

File: werkzeug_encryptedcookie/__init__.py [hunk 4]
```diff
@@ -62,6 +72,19 @@ def decrypt(cls, string, secret_key):
...
+        if not data.startswith(cls.compress_cookie_header):
+            return data
+
+        body = data[len(cls.compress_cookie_header):]
+        decompressed = brotli.decompress(body)
+        # cookie itself is empty dict `{}` at least
+        if decompressed not in (None, ''):
+            return decompressed
+
+        return data
```
Comment (homm) [inline werkzeug_encryptedcookie/__init__.py]: I believe it should be like this:

```python
    @classmethod
     def decompress(cls, data):
         if data.startswith(cls.compress_cookie_header):
             body = data[len(cls.compress_cookie_header):]
             try:
                  return brotli.decompress(body)
             except brotli.error:
                  pass
         return data
```

Comment (homm) [reply werkzeug_encryptedcookie/__init__.py]: Also, `brotli.decompress` can't return anything from `None`, `{}`, strings

File: werkzeug_encryptedcookie/__init__.py [hunk 5]
```diff
@@ -73,6 +96,12 @@ def unserialize(cls, string, secret_key):
 
         payload = cls.decrypt(string, secret_key)
 
+        if cls.compress_cookie:
```
[the remark this answers is not in the document]
Comment (andrewshkovskii) [reply werkzeug_encryptedcookie/__init__.py]: Done

**Pull request merged**